backend/src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -122,33 +148,71 @@
         });
     }
 
-    public Uni<Long> getEndpointsCountPerType(String accountId, Set<EndpointType> type, Boolean activeOnly) {
-        return sessionFactory.withSession(session -> {
-            return queryBuilderEndpointsPerType(accountId, type, activeOnly)
-                    .buildCount(session::createQuery)
-                    .getSingleResult();
+    public Uni<Long> getEndpointsCountPerCompositeType(String tenant, Set<CompositeEndpointType> type, Boolean activeOnly, boolean useStatelessSession) {
+        return commonStateSessionFactory.withSession(useStatelessSession, session -> queryBuilderEndpointsPerType(tenant, type, activeOnly)
+                .buildCount(session::createQuery)
+                .getSingleResult());
+    }
+
+    public Uni<Long> getEndpointsCountPerType(String tenant, Set<EndpointType> type, Boolean activeOnly, boolean useStatelessSession) {
+        return getEndpointsCountPerCompositeType(
+                tenant,
+                type.stream().map(CompositeEndpointType::new).collect(Collectors.toSet()),
+                activeOnly,
+                useStatelessSession
+        );
+    }
+
+    // Note: This method uses a stateless session
```

Comment:
All methods that rely only on the stateless session should be in `EndpointRepository` in the `engine`. Can you please move the code?

Reply:
Or maybe that method already exists there and needs to be updated. I'm checking.

Reply:
I think I added this method back because of the rebase

Reply:
Actually I think I made some updates - will double check.

Reply:
Looks like these changes were already applied to the repository